presto-hive-hadoop2/src/test/java/com/facebook/presto/hive/TestHiveFileSystemS3SelectPushdown.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.presto.hive.s3.HiveS3Config;
+import com.facebook.presto.hive.s3.PrestoS3ConfigurationUpdater;
+import com.facebook.presto.hive.s3.S3ConfigurationUpdater;
+import org.apache.hadoop.fs.Path;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Parameters;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.String.format;
+import static org.testng.util.Strings.isNullOrEmpty;
+
+public class TestHiveFileSystemS3SelectPushdown
```

Comment:
Thanks for taking care of this. 👍

Reply:
@sameer2086 @ranjanarajendran 
I wasn't aware that message relates to tests. Did you plan to capture logging output during test execution to verify S3 Select is being used? Is my understanding correct?

Reply:
I believe in the initial implementation of integration tests @ranjanarajendran performed assertions against the debug logs that were emitted with previous version of code. However, with the debug logs removed we don't have any way of asserting that S3SelectPushdown was indeed used. 

For the partial integration test, I manually introduced a bug on the S3SelectPushdown path and made sure the regression is caught. 

Thus, I was proposing to expose either some metrics or some post query stats in Presto's internal table to assert if a query used S3SelectPushdown. We are not familiar with that part of Presto's codebase so need some advise on how to proceed forward.

Reply:
Ideally, you would want to know not only that S3 Select was used, but also what exact condition has been pushed down.
Is it possible to ask S3 Select service for recent queries?

@electrum thoughts?

Reply:
Is it possible to run the integration tests against a mock S3 service that records the interaction?

Reply:
@findepi It is not possible to get recent queries from S3 Select. 

@electrum I believe these tests are for end to end integration test with real S3 Service. We could create another set of similar tests as a separate PR against Mock S3 Service to enable these assertions. However, as an end user, I still believe there is value in exposing this information through some stats or metrics. @nezihyigitbasi and I agreed that log is not the best place for this in earlier revision. Is there some other way through which we could expose this information for all queries?

Reply:
I believe the options are

* JMX metrics across all queries (e.g., `PrestoS3FileSystemStats`)
* Debug logging, which can contain information about specific queries

Reply:
@electrum, thanks for sharing your ideas.

* We will explore adding JMX metrics in later PRs.
* @nezihyigitbasi and I had previous discussion on setting up debug logs but Presto best practices do not permit for excessive logging Info or Debug. Thus we ended up removing the debug logs https://github.com/prestodb/presto/pull/11033#discussion_r234839553